### ISSUE TITLE
New Rule: no-empty-alt-text

### ISSUE THREAD

**Issue by Pixel998** [eslint/markdown#365]
### Rule details

The rule should warn when an image is missing alternate text

### What type of rule is this?

Warns about a potential problem

### Example code

Example of incorrect code:

```markdown
![]()
![][ref]

[ref]: image.jpg
```

### Participation

- [x] I am willing to submit a pull request to implement this rule.

### Additional comments

Rationale: Alternate text is important for accessibility and describes the content of an image for people who may not be able to see it.

**Comment by lumirlumir** [comment]
Hi @Pixel998, thanks for opening this issue!

It looks great to me, and I agree that this rule should be implemented as part of the recommended setup.

Interestingly, I’ve actually implemented this rule in my own custom ESLint plugin, `eslint-plugin-mark`, which I’m currently working on:

* https://eslint-plugin-mark.lumir.page/docs/rules/alt-text
* https://github.com/lumirlumir/npm-eslint-plugin-mark/blob/main/packages/eslint-plugin-mark/src/rules/alt-text/alt-text.js
* https://github.com/lumirlumir/npm-eslint-plugin-mark/blob/main/packages/eslint-plugin-mark/src/rules/alt-text/alt-text.test.js

So, if you don’t mind, and if other team members are also on board with including this rule, would it be alright if I took on this issue?

---

In return, I’d be happy to suggest a few other interesting ideas that could be useful for Markdown linting. These are all common mistakes found in Markdown writing:

* [Spaces inside emphasis markers](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md037---spaces-inside-emphasis-markers)
* [No space after hash on ATX-style headings](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md018---no-space-after-hash-on-atx-style-heading)

**Comment by nzakas** [comment]
I think this is a good idea. :+1: Let's let @Pixel998 work on this as they created the issue and indicated interest in implementing it.

**Comment by lumirlumir** [comment]
> Let's let @Pixel998 work on this as they created the issue and indicated interest in implementing it.

Sounds good. Let's do it.
